core/src/util/builders.cpp

```diff
@@ -59,22 +45,72 @@ void Builders::buildPolygon(const Polygon& _polygon, float _height, PolygonBuild
         }
     }
 
-    // call the tesselator
-    glm::vec3 normal(0.0, 0.0, 1.0);
+    _ctx.earcut(_polygon);
+
+    size_t sumPoints = 0;
+    for (auto& line : _polygon) {
+        sumPoints += line.size();
+    }
 
-    _ctx.numVertices += _ctx.earcut.vertices.size();
+    size_t sumVertices = 0;
+    _ctx.used.assign(sumPoints, 0);
+    for (auto i : _ctx.earcut.indices) {
```

Comment:
Ah ok, I'm beginning to understand the situation. Would've been quicker if there were comments ;)

Do we really need to determine which points are referenced by indices? The only situation I can imagine that would leave points unused is if a polygon has duplicated points, which should be rare enough that it's almost certainly faster to just add them to buffers anyway and not worry about it.